### DISCUSSION TITLE
A very long password is set to at least 16 characters. Where can I change this value to at least 8?

### DISCUSSION THREAD

**Discussion by nloshka404** [firefly-iii/firefly-iii#7958] [Firefly III]
A very long password is set to at least 16 characters. Where can I change this value to at least 8?

**Comment by nloshka404** [comment]
The new password value must be at least 16 characters.

**Comment by JC5** [comment]
You can't. This is hard coded and will not be an option.

**Reply by nloshka404** [reply]
Why such difficulties? It’s wildly inconvenient to use such long passwords; besides, for the paranoid, you can connect 2FA. For encoding, 8 characters are enough.
Do you also use 16-digit passwords everywhere? or maybe 32 is better?

**Comment by nloshka404** [comment]
Even Google does not require 16-digit passwords from its users!!!

**Comment by coolcow** [comment]
> Do you also use 16-digit passwords everywhere? or maybe 32 is better?

Probably none of your business and not a good strategy to be sarcastic when you request a feature for a product you are using for free.

---

> Why such difficulties? It’s wildly inconvenient to use such long passwords; besides, for the paranoid, you can connect 2FA.

I second that.

---

> Even Google does not require 16-digit passwords from its users!!!

Besides the unnecessary use of multiple exclamation marks: even most password safes don't require a password that long. 

---

I use an random password generator with the length set to 12. Having hundreds of accounts (many of them with sensitive stuff), firefly-iii is the first one where the password length wasn't enough to be accepted.
And that's for a reason: the security when using the full set of upper/lower-case, digit, and special characters is already pretty OK with only 8 characters for most uses. Sure, 10 or 12 would be even better and harder to bruteforce. But everything over that brings only theoretically more security but non in practice. It simply makes no difference if the password that couldn't be cracked in a human lifespan was 12 ore 1000 characters long.

It's even more annoying considering the fact, that my instance of firefly-iii runs only on my localhost, without any access from even other machines on the same network. That's why i can confess to you, that the password i currently use is `16characterslong`, and it is the unsafest password i use so far. But the security of this password is not an issue in my case because i know what i am doing and how much security i need. And that's exactly the point...

@JC5 i know you don't have to, but would you explain why the option to reduce the password length is such a big deal? One could put a big "DANGER" mark alongside the change button or something like that, with a short "do it on your own risk" explanation.
Maybe worth a reconsiding...

**Comment by sysosmaster** [comment]
> A.2 Length
Password length has been found to be a primary factor in characterizing password strength [[Strength]](https://pages.nist.gov/800-63-3/sp800-63b.html#strength) [[Composition]](https://pages.nist.gov/800-63-3/sp800-63b.html#composition). Passwords that are too short yield to brute force attacks as well as to dictionary attacks using words and commonly chosen passwords.
> 
> The minimum password length that should be required depends to a large extent on the threat model being addressed. Online attacks where the attacker attempts to log in by guessing the password can be mitigated by limiting the rate of login attempts permitted. In order to prevent an attacker (or a persistent claimant with poor typing skills) from easily inflicting a denial-of-service attack on the subscriber by making many incorrect guesses, passwords need to be complex enough that rate limiting does not occur after a modest number of erroneous attempts, but does occur before there is a significant chance of a successful guess.
> 
> Offline attacks are sometimes possible when one or more hashed passwords is obtained by the attacker through a database breach. The ability of the attacker to determine one or more users’ passwords depends on the way in which the password is stored. Commonly, passwords are salted with a random value and hashed, preferably using a computationally expensive algorithm. Even with such measures, the current ability of attackers to compute many billions of hashes per second with no rate limiting requires passwords intended to resist such attacks to be orders of magnitude more complex than those that are expected to resist only online attacks.
>
> Users should be encouraged to make their passwords as lengthy as they want, within reason. Since the size of a hashed password is independent of its length, there is no reason not to permit the use of lengthy passwords (or pass phrases) if the user wishes. Extremely long passwords (perhaps megabytes in length) could conceivably require excessive processing time to hash, so it is reasonable to have some limit.
[source: Nist 800-63-3(b)](https://pages.nist.gov/800-63-3/sp800-63b.html)

since we do not have a full SOC like google does, we have to use proper security means to keep users reasonably safe... passwords that are shorter than 16 characters, are not that... just learn to use Passphrases & password managers.

> i know you don't have to, but would you explain why the option to reduce the password length is such a big deal? One could put a big "DANGER" mark alongside the change button or something like that, with a short "do it on your own risk" explanation.
Maybe worth a reconsiding...

is not an option due to EU law, it would make @JC5 liable. (only safe and sensible options regarding security are allowed)

its also FOSS software, if you want a broken setup, you can break it yourself.

what could be done is implement something like a PassKey support. but I am not going to ask @JC5 to make it for me...

**Reply by coolcow** [reply]
> is not an option due to EU law, it would make @JC5 liable

How so ?

> only safe and sensible options regarding security are allowed

So why is `abcdefghigklmnop` or `16characterslong` allowed but `0d9v@&%*OK` isn't ? 

> its also FOSS software, if you want a broken setup, you can break it yourself.

That's exactly what I am going to do, except that the setup won't be broken but improved, with the 16 characters long requirement activated by default.

**Comment by JC5** [comment]
https://github.com/orgs/firefly-iii/discussions/7958#discussioncomment-6999164

**Reply by coolcow** [reply]
No discussion desired on this topic. OK.👍
